utils/responseUtil.ts

```diff
@@ -0,0 +1,51 @@
+import { NextApiResponse } from "next";
+
+/**
+ * @class ResponseUtil: Class for handling responses
+ */
+export class ResponseUtil {
+    /**
+     * Method for returning a 200 Status OK response
+     * @param res NextApiResponse
+     * @param data Optional data
+     */
+    static returnOK(res: NextApiResponse, data?: unknown): void {
+        res.status(200).json({ data });
+    }
+
+    /**
+     * Method for returning a status 400 Bad Request response
+     * @param res NextApiResponse
+     * @param message Optional message
+     */
+    static returnBadRequest(res: NextApiResponse, message: string): void {
+        res.status(400).json({ error: message });
+    }
+
+    /**
+     * Method for returning a status 404 Not Found Response
+     * @param res NextApiResponse
+     * @param message Optional message
+     */
+    static returnNotFound(res: NextApiResponse, message: string): void {
+        res.status(404).json({ error: message });
+    }
+
+    /**
+     * Method for returning a status 405 Method Not Allowed response
+     * @param res NextApiResponse
+     * @param allowedHeaders Array of Allowed Headers for Method
+     * @param message Optional message
+     */
+    static returnMethodNotAllowed(
+        res: NextApiResponse,
+        allowedHeaders: string[],
+        message?: string,
+    ): void {
+        if (message === undefined) {
+            message = "Method not allowed.";
+        }
+        res.setHeader("ALLOW", allowedHeaders);
```

Comment:
I don't know if this is important but I think this header needs to be "Allow"

Reply:
Will change